prac_07/project_management.py

```diff
@@ -0,0 +1,172 @@
+"""
+CP1404 Practical
+Estimated Time: 90 minutes
+Actual Time: 200 minutes
+"""
+from project import Project
+import datetime
+
+MENU = """- (L)oad projects
+- (S)ave projects
+- (D)isplay projects
+- (F)ilter projects by date
+- (A)dd new project
+- (U)pdate project
+- (Q)uit
+"""
+
+
+def main():
+    """Perform different functions based on user input"""
+    project_information = read_from_file()
+    print(MENU)
+    choice = input(">>> ").upper()
+
+    while choice != "Q":
+        if choice == "L":
+            project_information = read_from_file()
+            print("Projects have been loaded")
+
+        elif choice == "S":
+            save_to_file(project_information)
+            print("Projects have been saved")
+
+        elif choice == "D":
+            project_information = display_project_information(project_information, choice)
+
+        elif choice == "F":
+            filtered_project_information = []
+            date_string = input("Date (d/m/yyyy): ")  # e.g., "30/9/2022"
+            filter_date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+            for project in project_information:
+                if filter_date <= project.start_date:
+                    filtered_project_information += [project]
+            display_project_information(filtered_project_information, choice)
+
+        elif choice == "A":
+            new_project = add_new_project()
+            project_information.append(new_project)
+            print(f"Project: '{new_project.name}, "
+                  f"start: {new_project.start_date}, "
+                  f"priority = {new_project.priority}, "
+                  f"estimate: ${new_project.cost_estimate}, "
+                  f"completion: {new_project.completion_percentage}%' has been added")
+
+        elif choice == "U":
+            display_project_information(project_information, choice)
+            project_information, completed_projects, incomplete_projects = update_project_information(
+                project_information)
+
+        else:
+            print('Invalid menu choice')
+
+        print(MENU)
+        choice = input(">>> ").upper()
+
+    save_to_file(project_information)  # save to file when quitting
+    print("The file has been saved")
+    print("The program has ended")
+
+
+def display_project_information(project_information, choice):
+    """Print a sorted and formatted list of the projects"""
```

Comment:
This function does more then what is mentioned in the doc string. Try to split the function so it follows SRP.